### DISCUSSION TITLE
How to repair the DB

### DISCUSSION THREAD

**Discussion by josias-r** [directus/directus#3981] [Archive]
I've had a directus project running where I backed up the DB. I thought I can easily import the backup on a new DB and directus would be ready to go again.
Unfortunately, I was rather stupid and only backed the old DB up using CSV instead of an SQL dump.

Now after restoring I'm getting various errors in the application. I'm guessing some information like permissions, foreign keys etc. got lost during the CSV backup.

I tried both approaches where I imported first (which completely did not work) and the approach where I first ran `directus bootstrap` and then imported the data.

Is there some magic `directus repair db` command that can help me out? 😛

Or is there some way of exporting/importing the schema only of the collections?

**Comment by benhaynes** [comment]
No magic CLI command, no. CSV dumps are data only, right? Not sure if they even store the schema/datatypes.

**Reply by rijkvanzanten** [reply]
Yeah that's what I thought too!

@josias-r If you totally lost the schema information of that old DB, I think the nest best thing would be to configure Directus "fresh", and then import the CSV as just the data once that's done. Not ideal, but I don't think there's a way for Directus to just guess the schema based on the CSV data only 🤔

**Reply by josias-r** [reply]
Yup, data online. Thanks for the information.
In the meantime, I invested some time to recreate the collections manually. The `directus_fields` table export was really helpful, and with its content, I was able to remember all the necessary fields 😁
